### ISSUE TITLE
Empty contextPath is replaced with "/", causing jsp page compilation error with tomcat 7

### ISSUE THREAD

**Issue by tkent** [akhikhl/gretty#192]
__The short version__
Gretty currently swaps out empty contextPath strings for "/". When using tomcat 7 (at least 7.0.65), that causes a very specific bug where JSP compilation will break on the root servlet. You'll get an error about negative time on all JSP's.

Doing this swap is the wrong thing to do for tomcat, and passing an empty string as the contextPath fixes the issue. I have a patch for this and I can submit a pull request if you want. However, you might not want to fix the issue the way I have (see long version).

__The long version__
According to the [tomcat docs](https://tomcat.apache.org/tomcat-7.0-doc/config/context.html), the root context should be specified by an empty string (`""`). However, gretty currently fixes up contextPath strings that don't start with a "/" in [WebappConfig.prepareToRun()](https://github.com/tkent/gretty/blob/master/libs/gretty-core/src/main/groovy/org/akhikhl/gretty/WebAppConfig.groovy), by adding a "/" prefix. The result is you can't specify the `""` context path you need to.
 
The easy (and ignorable) symptom of this situation that tomcat spits on error along the lines of:

```
A context path must either be an empty string or start with a ''/'' and do not end with a ''/''...
```

However, the not-immediately-obvious symptom is that JSPs won't compile due to the "negative time" error I mentioned earlier.  Since this bug is very specific, I'm probably the only person to run into it.

--- Edit ---
The pull request fixes the issue by updating `TomcatServerConfigurer` to swap "/" to an empty string before setting the tomcat context name/path.

**Comment by ghar** [comment]
Great find - Running into the same issue.
I have been trying to figure out why farmRun was working on Jetty but not Tomcat.
Thanks

**Comment by tkent** [comment]
@ghar thanks! I figured I was the only person effected. Hopefully we can get the patch integrated soon.

**Comment by ghar** [comment]
@tkent If I fork your repo, is there a way for my build to pull from my fork on Github?

**Comment by tkent** [comment]
@ghar you can certainly use the patch right now.  Here's how you would do it.


1. Locally clone my repo
2. Checkout branch `t192`

__If you only care about using it locally:__

1. Publish the project to your local maven repository - `./gradlew publishToMavenLocal`

Providing you use mavenLocal() in your project that needs the patch, that's it. Next time you build your project with a --refresh-dependencies you'll pickup the new version and it will just work.

__If you want to share it amongst a team:__

You'll need to have access to an artifact repository that you can publish to, and that same repository will need to be used in the projects that need the patch. Providing you have that, then...

1. Update the `version` property in `gradle.properties` to something that makes sense. We used `1.2.5-t192-SNAPSHOT`
2. Modify `libs/common.gradle` so it publishes to the repo you have access to.

For us, since we don't happen use artifactory, we had to comment out the `apply plugin: 'com.jfrog.**' lines, remove the `bintray` and `artifactory` closures, then add a publishing closure specific for our repo.

good luck!

**Comment by tkent** [comment]
Closing this issue since the changes in my pull request (#194) have been integrated into the master branch. 

It looks like they will be available in this repository in gretty 1.4.0
